### PR TITLE
Support proving Cairo 0 programs

### PR DESCRIPTION
**Pull request by mellowcroc**

### CODE REVIEW

File: tests/test_run.rs [hunk 1]
```diff
@@ -256,7 +438,12 @@ fn test_run_e2e(
...
-    match run_cairo1(&prove_args, &tmp_dir) {
+    let run_result = match cairo_version {
+        CairoVersion::cairo0 => run_cairo0(&prove_args, &tmp_dir).map_err(|e| anyhow::anyhow!(e)),
+        CairoVersion::cairo1 => run_cairo1(&prove_args, &tmp_dir),
+    };
+
+    match run_result {
         Ok(_) => {
             let filename = program_file.file_stem().unwrap().to_str().unwrap();
             let air_public_input = tmp_dir
```
Comment (rot256) [inline tests/test_run.rs:449]: There is a lot of nested error handling around here.
Consider just doing result.expect("XYZ went wrong") throughout.

File: src/cairo.rs [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+
+    let air_private_input_path = tmp_dir
+        .path()
+        .join(format!("{}_air_private_input.json", filename));
+    let trace_absolute_path = trace_path
+        .as_path()
+        .canonicalize()
+        .unwrap_or(trace_path.clone())
+        .to_string_lossy()
+        .to_string();
```
Comment (rot256) [inline src/cairo.rs:126]: Why not use [.to_str](https://doc.rust-lang.org/stable/std/path/struct.PathBuf.html#method.to_str)? Do we want it to be lossy or fail?

**Review by rot256 — APPROVED** [review]
In the `examples/` directory, there is a bunch of symlinks to cairo programs in the root, but where are the actual files?

Otherwise looks good to me.

File: tests/test_run.rs (hunk 1 above)
Comment (mellowcroc) [reply tests/test_run.rs]: Done

**Comment by mellowcroc** [comment]
> In the `examples/` directory, there is a bunch of symlinks to cairo programs in the root, but where are the actual files?
> 
> Otherwise looks good to me.
Symlinks were copied from the `cairo-vm`, so they're invalid. Thought I removed these but alas 😅

File: src/cairo.rs (hunk 2 above)
Comment (mellowcroc) [reply src/cairo.rs]: Unintended change. Fixed.

**Pull request merged**